minimint-api/src/encoding/mod.rs

```diff
@@ -215,6 +217,22 @@ impl Decodable for String {
     }
 }
 
+impl Encodable for lightning_invoice::Invoice {
+    fn consensus_encode<W: Write>(&self, writer: W) -> Result<usize, Error> {
+        self.to_string().consensus_encode(writer)
+    }
+}
+
+impl Decodable for lightning_invoice::Invoice {
+    fn consensus_decode<D: Read>(d: D) -> Result<Self, DecodeError> {
+        let invoice_string =
+            String::from_utf8(Decodable::consensus_decode(d)?).map_err(DecodeError::from_err)?;
```

Comment:
You can directly decode strings, no need to go via `Vec<u8>`

Reply:
So something like this I think ...

```
String::consensus_decode(d)?
    .parse::<lightning_invoice::Invoice>()
    .map_err(DecodeError::from_err)
```